src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  get isAlive() {
+    return this.health > 0;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.cleanupDeadAnimals();
+    }
+  }
 }
 
+Animal.alive = [];
+Animal.cleanupDeadAnimals = () => {
+  Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore
+      && !animal.hidden
+      && animal.isAlive
```

Comment:
The 'bite' method correctly checks if the target animal is an instance of 'Herbivore' and not hidden, as well as if it's alive before applying damage. This is in line with the checklist recommendations. However, the check for 'animal.isAlive' is redundant here since the 'applyDamage' method inside the 'Herbivore' class already handles the case when health becomes less than or equal to 0 by calling 'cleanupDeadAnimals'. You can remove the 'animal.isAlive' check to simplify the condition.